gammapy/detect/cwt.py

```diff
@@ -57,200 +62,552 @@ def difference_of_gauss_kernel(radius, scale_step, n_sigmas=8):
+        Return cwt data from all the iterations.
+
+        Returns
+        -------
+        data : list
+            List of '~gamapy.detect.CWTData' objects.
+        """
+        if self.history is None:
+            raise AttributeError('This CWT object does not support saving cwt data images.')
+        else:
+            return copy.deepcopy(self.history)
+
+
+class CWTKernels(object):
+    """
+    Conduct arrays of kernels and scales for CWT algorithm.
+
+    Parameters
+    ----------
+    n_scale : int
+        Number of scales.
+    min_scale : double
+        First scale used.
+    step_scale : double
+        Base scaling factor.
+    old : boolean (default True)
+        DEBUG attribute. If False, use astropy kernels.
+
+    Attributes
+    ----------
+    n_scale : int
+        Number of scales considered.
+    min_scale : double
+        First scale used.
+    step_scale : double
+        Base scaling factor.
+    scales : '~numpy.ndarray'
+        Grid of scales.
+    kern_base : dict
+        Dictionary of scale powers as keys and 2D kernel arrays.
+        (mexican hat) as values
+    kern_approx : '~numpy.ndarray'
+        2D Gaussian kernel array from maximum scale.
+
+    Examples
+    --------
+    >>> from gammapy.detect import CWTKernels
+    >>> kernels = CWTKernels(n_scale=2, min_scale=6.0, step_scale=1.3)
+    >>> print (kernels.scales)
+    [ 6.   7.8]
+    """
+    def __init__(self, n_scale, min_scale, step_scale, old=True):
+        self.n_scale = n_scale
+        self.min_scale = min_scale
+        self.step_scale = step_scale
+        self.scales = np.array([min_scale * step_scale ** _ for _ in range(n_scale)],
+                               dtype=float)
+
+        self.kern_base = dict()
+        for idx_scale, scale in enumerate(self.scales):
+            if old:
+                self.kern_base[idx_scale] = difference_of_gauss_kernel(scale, step_scale)
+            else:
+                # maybe instantly create scales = self.scales * step_scale ?
+                self.kern_base[idx_scale] = MexicanHat2DKernel(scale * step_scale).array
+
+        max_scale = min_scale * step_scale ** n_scale
+        if old:
+            self.kern_approx = gauss_kernel(max_scale)
+        else:
+            self.kern_approx = Gaussian2DKernel(max_scale).array
+
+    def info(self):
+        """
+        Print information about scales and kernels of the object.
+
+        Returns
+        -------
+        information : string
+            Information about the object.
+        """
+        info_list = []
+        info_list.append('CWTKernels object')
+        info_list.append('-----------------')
+        info_list.append('Number of scales: {0}'.format(self.n_scale))
+        info_list.append('Minimal scale: {:.2f}'.format(self.min_scale))
+        info_list.append('Step scale: {:.2f}'.format(self.step_scale))
+        info_list.append('Scales: {0}'.format(str(self.scales)))
+        # TODO: add information about the kernels.
```

Comment:
Can you print the kernel sizes here?
I presume it's always the same size in x and y, i.e. it's enough to print the "width" of the kernel?

We always use this style in Gammapy to make info strings:
```
ss = 'spam\n'
ss += 'spam spam spam\n'
...
return ss
```
Please do that here as well.
(efficiency of this method is not a concern at all, it's very fast anyways)